packages/orchestration/src/utils/address.js

```diff
@@ -34,6 +34,20 @@ export const makeICAConnectionAddress = (
   });
   return `/ibc-hop/${controllerConnectionId}/ibc-port/icahost/${ordering}/${connString}`;
 };
+harden(makeICAChannelAddress);
+
+/**
+ * @param {IBCConnectionID}  controllerConnectionId
+ * @param {{ version?: string }} [opts]
+ */
+export const makeICQChannelAddress = (
+  controllerConnectionId,
+  { version = 'icq-1' } = {},
```

Comment:
is it OK for version to be any string at all, including those containing `/`?
If so, I'd appreciate a test to say so explicitly.

Reply:
Good callout. I've added [0c86102](https://github.com/Agoric/agoric-sdk/pull/9198/commits/0c861022f878393fcdd95a449cad3ea18dd18beb) which documents the limited functionality better.  This function will not be exposed to API consumers, so I didn't feel it necessary to validate `version`. As I'm writing this comment it feels like a good idea to add the `@internal` typedoc annotation.

Doing some additional testing, it seems like `network` / `ibc` do not throw if we provide an invalid remoteAddr to `.connect()`.  Is this worth implementing? Looks like [`decodeRemoteIbcAddress`](https://github.com/Agoric/agoric-sdk/blob/edbcace516bd1e51692b59533036256422979853/packages/vats/tools/ibc-utils.js#L11-L17), or at least the part that throws a `TypeError`, used in the `onConnect` handler could be leveraged in [`connect`](https://github.com/Agoric/agoric-sdk/blob/ae0e43ea09c4a87d7fe49dac6cb21920182a5669/packages/network/src/network.js#L494-L510) cc @michaelfig

Reply:
Ok cool. Attempted this here: [733bbe5](https://github.com/Agoric/agoric-sdk/pull/9198/commits/733bbe5f2754221b297a91337a6bcb4fd75ade58), please lmk what you think.

I was thinking to make `ProtocolHandler['validateRemoteAddr']` optional, but wasn't sure the best way to verify the existence of a remote method (i.e. `if (protocolHandler.validateRemoteAddr) protocolHandler.validateRemoteAddr()`)

Reply:
> ... wasn't sure the best way to verify the existence of a remote method

The only way is to `try` it.

Reply:
> The only way is to `try` it.
Yes, and if you see how `rethrowUnlessMissing` is used as a `catch` handler, that will interpret the rejection and if it seems to be due to a missing method, it will swallow the error, otherwise will `throw` it again.

Reply:
`rethrowUnlessMissing` - nice! was going to say it'd be handy to have a function/tool for this